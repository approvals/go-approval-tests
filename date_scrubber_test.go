package approvals_test

import (
	"fmt"
	"testing"

	approvals "github.com/approvals/go-approval-tests"
)

func TestSupportedFormatWorksForExamples(t *testing.T) {
	for _, supportedFormat := range approvals.GetSupportedFormats() {
		dateScrubber := approvals.NewDateScrubber(supportedFormat.Regex)
		for _, example := range supportedFormat.Examples {
			result := dateScrubber(example)
			if result != "[Date1]" {
				t.Errorf("did not work for\nregex: %s\nexample: %s\ngot: %s", supportedFormat.Regex, example, result)
			}
		}
	}
}

func TestGetDateScrubber(t *testing.T) {
	formats := approvals.GetSupportedFormats()
	output := ""
	for _, format := range formats {
		for _, example := range format.Examples {
			scrubber, err := approvals.GetDateScrubberFor(example)
			if err != nil {
				t.Error(err)
			}
			exampleText := fmt.Sprintf("{'date':\"%s\"}", example)
			result := scrubber(exampleText)
			expected := fmt.Sprintf("Scrubbing for %s:\nExample: %s\n\n", example, result)
			output += expected
		}
	}
	approvals.VerifyString(t, output)
}

func TestExampleForDocumentation(t *testing.T) {
	// begin-snippet: scrub-date-example
	scrubber, err := approvals.GetDateScrubberFor("00:00:00")
	if err != nil {
		t.Error(err)
	}
	approvals.VerifyString(t, "created at 03:14:15", approvals.Options().WithScrubber(scrubber))
	// end-snippet
}

//   @Test
//   void supportedFormats()
//   {
//     VelocityApprovals.verify(c -> c.put("formats", DateScrubber.getSupportedFormats()),
//         new Options().forFile().withExtension(".md"));
//   }
