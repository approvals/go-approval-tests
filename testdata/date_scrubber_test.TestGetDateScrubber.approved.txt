Scrubbing for Tue May 13 16:30:00:
Example: {'date':"[Date1]"}

Scrubbing for Wed Nov 17 22:28:33 EET 2021:
Example: {'date':"[Date1]"}

Scrubbing for Wed, 21 Oct 2015 07:28:00 GMT:
Example: {'date':"[Date1]"}

Scrubbing for Tue May 13 2014 23:30:00.789:
Example: {'date':"[Date1]"}

Scrubbing for Tue May 13 16:30:00 -0800 2014:
Example: {'date':"[Date1]"}

Scrubbing for 13 May 2014 23:50:49,999:
Example: {'date':"[Date1]"}

Scrubbing for Oct 13 15:29:
Example: {'date':"[Date1]"}

Scrubbing for May 13, 2014 11:30:00 PM PST:
Example: {'date':"[Date1]"}

Scrubbing for 23:30:00:
Example: {'date':"[Date1]"}

Scrubbing for 2014/05/13 16:30:59.786:
Example: {'date':"[Date1]"}

Scrubbing for 2014/05/13 16:30:59:
Example: {'date':"[Date1]"}

Scrubbing for 2020-9-10T08:07Z:
Example: {'date':"[Date1]"}

Scrubbing for 2020-09-9T08:07Z:
Example: {'date':"[Date1]"}

Scrubbing for 2020-09-10T8:07Z:
Example: {'date':"[Date1]"}

Scrubbing for 2020-09-10T08:07Z:
Example: {'date':"[Date1]"}

Scrubbing for 2020-09-10T08:07:89Z:
Example: {'date':"[Date1]"}

Scrubbing for 2020-09-10T01:23:45.678Z:
Example: {'date':"[Date1]"}

Scrubbing for 20210505T091112Z:
Example: {'date':"[Date1]"}

Scrubbing for 2024-12-17:
Example: {'date':"[Date1]"}

Scrubbing for 2024-12-18T14:04:46.746130Z:
Example: {'date':"[Date1]"}

Scrubbing for 2024-12-18T14:04:46Z:
Example: {'date':"[Date1]"}

Scrubbing for 2024-12-18T14:04:46.746130834Z:
Example: {'date':"[Date1]"}

Scrubbing for 13/05/2014 23:50:49:
Example: {'date':"[Date1]"}

Scrubbing for 13.05.2014 23:50:49:
Example: {'date':"[Date1]"}

Scrubbing for 13-05-2014 23:50:49:
Example: {'date':"[Date1]"}

Scrubbing for 13.05.2014 23:50:
Example: {'date':"[Date1]"}

Scrubbing for 05/13/2014 11:50:49 PM:
Example: {'date':"[Date1]"}

